CMakeLists.txt

```diff
@@ -79,15 +79,13 @@ set(FLAGS_ALL "-fopenmp-simd")  # enables loop vectorization hints, but does not
 set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} ${FLAGS_ALL}")
 set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} ${FLAGS_ALL}")
 
-# Build the binary optimized for the current system, ignoring older systems. For x86 systems, we use -march=native.
-# Clang does not support -march=native on ARM. ARM recommends setting -mcpu for both GCC and Clang instead of march:
-# https://community.arm.com/developer/tools-software/tools/b/tools-software-ides-blog/posts/compiler-flags-across-architectures-march-mtune-and-mcpu
-# We check specifically for arm64 (e.g., Apple's M1) or aarch64 (e.g., Raspberry Pi 4), as these are the two ARM architectures we have tested so far.
-# Match for "arm" on your own risk to include other ARM architures.
-if (CMAKE_SYSTEM_PROCESSOR MATCHES "arm64|aarch64")
-    set(FLAGS_RELEASE "-mcpu=native")
-else()
-    set(FLAGS_RELEASE "-march=native")  
+# Build the binary optimized for the current system, ignoring older systems.
+# Not all environments support march=native - check before we use it.
+set(FLAGS_RELEASE "-mcpu=native -mtune=native")
```

Comment:
The reason for not setting all three was the following summary of the linked article (it's an ARM article):
```
TL;DR: Whenever possible use only -mcpu=native. Avoid -march and -mtune.
```

Reply:
My understand is that this summary is limited to ARM. As far as I know, march is the way to go for x86. That contradicts the tldr.

Reply:
Hmm, what you said has been done before this PR. Why does this contradict the tldr?

Reply:
Are you talking about the addition of mtune? As far as I understand it, it doesn't hurt, but `mcpu` is more specific in some cases.